### ISSUE TITLE
Failed to connect when setting up Uptime Kuma integration

### ISSUE THREAD

**Issue by mploetner** [meichthys/uptime_kuma#99]
I had the custom Uptime Kuma integration from HACS running and from one day to another it stopped. I reinstalled it and when I enter my uptime Kuma credentials (ip, port, user etc.) it says „Failed to connect“ I set correctly to ignore SSL and added http:// in front of the IP (as before). Uptime Kuma is running and I can access it via the IP

I also tried to set another Uptime Kuma instance which I was able to integrate and had it running for 2 hour and then the integration stopped working while the Kuma instance itself works flawlessly. Trying to reconnect I get the same error as with the other instance.

**Comment by dieuvolkanos** [comment]
Same error for me. Also tried with API keys and unable to connect.

**Comment by dandek01** [comment]
Same error here. This all begin after updating to home assistant 2024.5
This is the log
```
2024-05-02 21:33:54.032 ERROR (MainThread) [custom_components.uptime_kuma] Request timeout for 'https://uptimekuma.myserver.com/:443/metrics'
2024-05-02 21:34:24.909 ERROR (MainThread) [custom_components.uptime_kuma] Request exception for 'uptimekuma.myserver.com:443/metrics' with - uptimekuma.myserver.com:443/metrics
```
if i try to open link in browser uptimekuma.myserver.com:443/metrics i cant log in. Username and password is correct. I can log in with https://uptimekuma.myserver.com and i see metrics when I'm log in.
I have similar problem with adguard and frigate.

**Comment by countrock** [comment]
Using the HACS plugin on port 3001. Having the issue of Failed to connect. Even disabled authentication, API or user creds. All have the same issue.

**Comment by dandek01** [comment]
I'm using ngnix with letsencrypt in docker. I bought a domain and configured some of the containers to use subdomans like https://uptimekuma.myserver.com/.
Certs are renewed automatically every few months. I just use adguard dns rewrite *.myserver.com to point back everything to my local server ip.
And this worked for a year or two without any problems. After updating ha 2024.5 first my met.no integration broke down, followed by local tuya uptime kuma, frigate and ad guard. Last three were configured to use ssl and everything worked.
The only thing I noticed with uptime kuma is this error in logs

```
This error originated from a custom integration.

Logger: custom_components.uptime_kuma
Source: custom_components/uptime_kuma/config_flow.py:60
integration: uptime_kuma (documentation, issues)
First occurred: 21:03:07 (2 occurrences)
Last logged: 23:31:36

Request timeout for 'https://uptimekuma.my.com:443/metrics'
```

If I open that url in browser I cant log in as it is asking for credentials over and over again. The same thing is if I try http address. But this is my username and password for uk. 
Tried bunch of things that cross my mind and ended up with this.
But this is my username and password for uk. I don't know. Maybe it is me but I think that this is somehow related to last update, or maybe my setup, but I still cant set up integrations using ssl to uk, frigate and adguard.

**Comment by dandek01** [comment]
I'm using ver. 1.23.13
This is my docker-compose file:
```
  uptime-kuma:
    image: louislam/uptime-kuma:1
    container_name: uptime-kuma
    environment:
      - TZ=Europe/Zagreb
    volumes:
      - ./uptime-kuma-data:/app/data
      - /run/docker.sock:/run/docker.sock
    ports:
      - 3001:3001  
    restart: always
```
I can log into dashboard with credentials using both http and https.

**Comment by dandek01** [comment]
I reviewed my nginx settings. I created access list to allow connections only from my local ips. I removed that from all containers I put it on and now I can access and log in using api on host https://uptimekuma.mysever.com/metrics
When I tried to set up integration I got this address 
```
https://uptimekuma.myserver.com/:443/metrics
```
Note "/" before :433. If I try to open this page in browser I got page not found. My suspicions is on nginx config file for uptime kuma, but I'm not sure. I tied to set it up without "/" at the end but the same thing. It tries to connect to non existent url. 
This is config file for uk in nginx.


```
# ------------------------------------------------------------
# uptimekuma.myserver.com
# ------------------------------------------------------------

map $scheme $hsts_header {
    https   "max-age=63072000; preload";
}

server {
  set $forward_scheme http;
  set $server         "192.168.31.103";
  set $port           3001;

  listen 80;
listen [::]:80;

listen 443 ssl http2;
listen [::]:443 ssl http2;

  server_name uptimekuma.myserver.com;

  # Let's Encrypt SSL
  include conf.d/include/letsencrypt-acme-challenge.conf;
  include conf.d/include/ssl-ciphers.conf;
  ssl_certificate /etc/letsencrypt/live/npm-20/fullchain.pem;
  ssl_certificate_key /etc/letsencrypt/live/npm-20/privkey.pem;

# Asset Caching
  include conf.d/include/assets.conf;

  # HSTS (ngx_http_headers_module is required) (63072000 seconds = 2 years)
  add_header Strict-Transport-Security $hsts_header always;

    # Force SSL
    include conf.d/include/force-ssl.conf;

proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $http_connection;
proxy_http_version 1.1;

  access_log /data/logs/proxy-host-11_access.log proxy;
  error_log /data/logs/proxy-host-11_error.log warn;

  location / {

  # HSTS (ngx_http_headers_module is required) (63072000 seconds = 2 years)
  add_header Strict-Transport-Security $hsts_header always;

    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_http_version 1.1;
    
    # Proxy!
    include conf.d/include/proxy.conf;
  }

  # Custom
  include /data/nginx/custom/server_proxy[.]conf;
}
```
This config is generated using nginx webui.

**Comment by dandek01** [comment]
I tied to turn on/off additional options, restore it from backup, rebuild image. Nothing. The only thing I didnt try is to delete container, it's image, local files, pull it from docker hub and reconfigure everything from scratch.

**Comment by dandek01** [comment]
I think that problem is the same for uptime kuma, frigate and adguard integration connecting them over ssl.
If I want to connect frigate over ssl I get timeout error fetching https://frigate.myserver.com/api/stats but I can access it in the browser private mode.
But I dont know what the problem is.

**Comment by dandek01** [comment]
No. I tried to add uptime kuma from hacs repository but ha wont let me add it as this integration is already built in home assistant.

**Comment by dandek01** [comment]
I can' add uptime kuma integration from hacs as it is already added in ha. Tried that.
I use docker compose for ha and all others "addons".

**Comment by dandek01** [comment]
Oh sorry gave you wrong info. Was working on something else. 
Yes, I have integration v.2.30  from HACS.

**Comment by dieuvolkanos** [comment]
After many try this configuration working for me.

Create an API Key on Kuma

[Uptime Kuma Integration]
Host: http://<ip>
port: 3001
Username: Username
Password: API Key

**Comment by dandek01** [comment]
I tried and it created integration. I just used homeassistant.local.
I doesnt make much sense to me. I have glances integration that is using https and it is working, no problem.

**Comment by dandek01** [comment]
I'm not running addon, I'm running uptime kuma in docker container. And everything else for that matter. I can stop homeassistant container and uptime kuma is.. up with all 47 monitors.
I mostly use it to check zigbee devices availability.  Everything is reported online even if homeassistant is stopped.

**Comment by dandek01** [comment]
Everything is running on the same machine. 31 container. I use https on local network. I know it is wired unnecessary etc. but this is how I set things up. Was just checking containers and glances container is running in network mode host and I can connect integration using ssl. Others like frigate, adguard or uptime kuma have mapped ports and this is a bridge mode if I got docker networking correctly. As this was working fine before I presume that something has been changed in home assistant that this isn't working anymore.
But I will have to use host on some of this containers to test things out.

**Comment by dandek01** [comment]
I solve it. The problem was dns settings on my debian machine. Probably years back I set up 1.1.1.1 nameserver. When I done curl -v https://frigate.myserver.com it tried to access it over my public ip, but containers can't be accessed over the net. And that was the reason I coudn't set uo integrations using mysubdomain over ssl.
I set up frigate and uptime kuma integrations to work with ssl, still struggling with adguard over ssl.
